## 6. Analisis Statistik Deskriptif
Bagian ini menghitung ukuran pemusatan, penyebaran, dan letak data pada variabel cuaca dan polusi untuk memahami distribusi data secara keseluruhan.

In [4]:
import pandas as pd
df = pd.read_csv("data/air_quality.csv")

kolom_pm25 = df["pm25"]
kolom_temp = df["temp"]
kolom_no2 = df["no2"]

print("=== 1. UKURAN PEMUSATAN (PM2.5) ===")
print("Rata-rata (Mean)       :", kolom_pm25.mean())
print("Nilai Tengah (Median)  :", kolom_pm25.median())
print("Sering Muncul (Modus)  :", kolom_pm25.mode()[0])

print("\n=== 2. UKURAN PENYEBARAN (PM2.5) ===")
print("Nilai Terkecil (Min)   :", kolom_pm25.min())
print("Nilai Terbesar (Max)   :", kolom_pm25.max())
print("Jangkauan (Range)      :", kolom_pm25.max() - kolom_pm25.min())
print("Varians                :", kolom_pm25.var())
print("Simpangan Baku (Std)   :", kolom_pm25.std())

print("\n=== 3. UKURAN LETAK (PM2.5) ===")
Q1 = kolom_pm25.quantile(0.25)
Q3 = kolom_pm25.quantile(0.75)
IQR = Q3 - Q1
print("Kuartil 1 (Q1)         :", Q1)
print("Kuartil 3 (Q3)         :", Q3)
print("Jangkauan Antar Kuartil (IQR):", IQR)

print("\n=== RINGKASAN DESCRIBE (3 KOLOM) ===")
print(df[["pm25", "temp", "no2"]].describe())

print("\n=== [NILAI PLUS] HITUNG MANUAL SIMPANGAN BAKU PM2.5 (10 BARIS PERTAMA) ===")
data_10_baris = df["pm25"].head(10)
rata_rata_10 = data_10_baris.mean()
selisih_kuadrat = ((data_10_baris - rata_rata_10) ** 2).sum() # Rumus: Jumlah (x - rata-rata)^2 dibagi (n-1)
varians_manual = selisih_kuadrat / (len(data_10_baris) - 1)
std_manual = varians_manual ** 0.5 # Simpangan baku adalah akar dari varians

print("Hasil Manual Rumus     :", std_manual)
print("Hasil Fungsi Pandas    :", data_10_baris.std())
print("Kesimpulan             : Hasil manual dan pandas sama persis!")

=== 1. UKURAN PEMUSATAN (PM2.5) ===
Rata-rata (Mean)       : 57.82832615310326
Nilai Tengah (Median)  : 47.37921975110725
Sering Muncul (Modus)  : 15.0

=== 2. UKURAN PENYEBARAN (PM2.5) ===
Nilai Terkecil (Min)   : 15.0
Nilai Terbesar (Max)   : 926.1494770860722
Jangkauan (Range)      : 911.1494770860722
Varians                : 2165.8344710341935
Simpangan Baku (Std)   : 46.53852673897395

=== 3. UKURAN LETAK (PM2.5) ===
Kuartil 1 (Q1)         : 20.735777427980317
Kuartil 3 (Q3)         : 81.71994786581891
Jangkauan Antar Kuartil (IQR): 60.98417043783859

=== RINGKASAN DESCRIBE (3 KOLOM) ===
               pm25          temp           no2
count  87672.000000  87672.000000  87672.000000
mean      57.828326     26.967928     20.380443
std       46.538527      9.144992     12.095623
min       15.000000     10.000000      5.000000
25%       20.735777     20.158294     11.487912
50%       47.379220     26.484439     18.405700
75%       81.719948     33.506717     26.568068
max      926.149

## Interpretasi Statistik Deskriptif Variabel Numerik

Berdasarkan perhitungan statistik deskriptif di atas, berikut adalah penjelasan untuk sebaran datanya:
1. **Perbandingan Rata-rata dan Median (PM2.5):** Nilai rata-rata polusi udara adalah **57.8 µg/m³**, sedangkan nilai tengahnya (median) berada di **47.3 µg/m³**. Karena nilai rata-rata lebih besar dibandingkan median, ini berarti bentuk sebaran datanya miring ke kanan (*right-skewed*). Artinya, kualitas udara sebagian besar waktu berada di angka yang rendah, namun ada beberapa kejadian ekstrem di mana polusi melonjak sangat tinggi (hingga maksimal 926.1 µg/m³).
2. **Ukuran Penyebaran (Varians dan Simpangan Baku):** Simpangan baku PM2.5 sebesar **46.5 µg/m³** dan suhu udara sebesar **9.1 °C** menunjukkan seberapa bervariasinya cuaca dan polusi dari rata-ratanya setiap jam.
3. **Ukuran Letak:** Sebanyak 50% data inti kualitas udara berada di dalam Jangkauan Antar Kuartil (IQR), yakni antara 20.7 µg/m³ (Q1) hingga 81.7 µg/m³ (Q3).

## 7. Analisis Variabel Kategorik
Oleh karena seluruh data mentah bersifat numerik kontinu, nilai ambang batas polusi dikonversi menjadi kategori diskret guna mempermudah evaluasi persentase tingkat bahaya.

In [5]:
# Membuat fungsi untuk mengubah angka PM2.5 menjadi kategori teks
def kriteria_polusi(nilai):
    if nilai <= 50:
        return "Aman"
    elif nilai <= 100:
        return "Sedang"
    else:
        return "Berbahaya"

# Membuat kolom kategorik baru bernama 'kategori_polusi'
df["kategori_polusi"] = df["pm25"].apply(kriteria_polusi)

print("=== TABEL FREKUENSI KATEGORI POLUSI ===")
print(df["kategori_polusi"].value_counts())

print("\n=== TABEL PERSENTASE KATEGORI (%) ===")
print(df["kategori_polusi"].value_counts(normalize=True) * 100)

=== TABEL FREKUENSI KATEGORI POLUSI ===
kategori_polusi
Aman         45878
Sedang       27800
Berbahaya    13994
Name: count, dtype: int64

=== TABEL PERSENTASE KATEGORI (%) ===
kategori_polusi
Aman         52.329136
Sedang       31.709098
Berbahaya    15.961767
Name: proportion, dtype: float64


## Ringkasan Variabel Kategorik

Oleh karena seluruh data metrik cuaca berbentuk angka, kami mengelompokkan nilai numerik $PM_{2.5}$ menjadi sebuah variabel kategorik baru bernama **Kategori Polusi** (Aman, Sedang, Berbahaya).

Berdasarkan tabel frekuensi di atas, dapat disimpulkan bahwa:
- **Modus (Kategori paling sering muncul):** Adalah kategori **"Aman"**.
- Secara persentase, sebagian besar waktu di kota tersebut memiliki udara yang aman, meskipun frekuensi kategori "Berbahaya" (di atas 100 µg/m³) tetap sering terjadi. Pengelompokan ini akan memudahkan analisis lanjutan mengenai kapan polusi mencapai titik tidak sehat.